appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt

```diff
@@ -3,14 +3,61 @@ package org.kinecosystem.appsdiscovery.sender.discovery.view
 import android.content.Context
 import android.content.Intent
 import android.os.Bundle
+import android.support.constraint.Group
 import android.support.v7.app.AppCompatActivity
+import android.view.View
+import android.widget.ProgressBar
 import org.kinecosystem.appsdiscovery.R
+import org.kinecosystem.appsdiscovery.sender.discovery.presenter.AppsDiscoveryListPresenter
+import org.kinecosystem.appsdiscovery.sender.discovery.view.customView.AppsDiscoveryList
 
 class AppsDiscoveryActivity : AppCompatActivity() {
 
+    lateinit var dataGroup: Group
+    lateinit var noDataGroup: Group
+    lateinit var loader: ProgressBar
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.apps_discovery_activity)
+        dataGroup = findViewById(R.id.data)
+        noDataGroup = findViewById(R.id.noData)
+        loader = findViewById(R.id.loader)
+        findViewById<AppsDiscoveryList>(R.id.list).setLoadingListener(object : AppsDiscoveryListPresenter.LoadingListener {
+            override fun loading() {
+                updateLoading()
```

Comment:
here by calling updateLoading here you actually add the activity and it's views as a member variable of your listener.  You pass this listener to a presenter and model that might outlive the activity and thus continue holding a reference to it after it has already been destroyed. Therefore you need to make sure you remove the listener when activity is destroyed.

Reply:
👍

Reply:
berry if the activity is destroyed then the list is detached?
if hold ref to the activity but its a view that has a ref to the parent activity
what do you think?

Reply:
i reset the listener to null when detached